Update doc for generate modular sdk (#34057)
Azure/azure-sdk-for-js: documentation/Generate-code-from-TypeSpec.md

### How to configure tspconfig.yaml
You can reference these two config files to configure the Modular or RLC package:

> - [Modular tspconfig.yaml for MPG](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml)

Reviewer: ```suggestion
- [Modular tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml)
```

- [RLC tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/ai/Face/tspconfig.yaml)

Please make sure `service-dir`, `package-dir`, `package-details`(for typespec-ts) is correctly configured. `experimental-extensible-enums` is the optional config.

> For MPG ,`is-modular-library` default behavior is true. If you want to generate modular sdk for DPG, you should add

Reviewer: ```suggestion
For control-plane SDKs, the `is-modular-library` option is true by default, while for data-plane SDK it is false. If you want to generate Modular libraries for data-plane SDKs, you should add
```

### How to configure tspconfig.yaml
You can reference these two config files to configure the Modular or RLC package:
- [Modular tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml)
- [RLC tspconfig.yaml](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml)

Please make sure `service-dir`, `package-dir`, `package-details`, `flavor`(for typespec-ts) is correctly configured. `experimental-extensible-enums` is the optional config.

> For control-plane SDKs, the `is-modular-library` option is true by default, while for data-plane SDKs it is false. If you want to generate Modular libraries for data-plane SDKs, you should add

Reviewer: ```suggestion
For control-plane SDKs, the `is-modular-library` option is true by default, while for data-plane SDKs it is false. If you want to generate Modular libraries for data-plane SDKs and you need to get architects approval for that, then you should add
```
